src/main/flight/gps_rescue.c

```diff
@@ -238,48 +236,11 @@ static void rescueAttainPosition(void)
     /**
         Altitude (throttle) controller
     */
-    const float altitudeErrorCm = (rescueState.intent.targetAltitudeCm - getAltitudeCm());
-    // height above target in cm (negative means too low)
-    // at the start, the target starts at current altitude plus one step.  Increases stepwise to intended value.
-
-    // P component
-    const float throttleP = getAltitudePidCoeffs()->Kp * altitudeErrorCm;
-
-    // I component
-    // reduce the iTerm gain for errors greater than 2m, otherwise it winds up too much
-    const float itermNormalRange = 200.0f; // 2m
-    const float itermRelax = (fabsf(altitudeErrorCm) < itermNormalRange) ? 1.0f : 0.1f;
-
-    throttleI += altitudeErrorCm * getAltitudePidCoeffs()->Ki * itermRelax * taskIntervalSeconds;
-    throttleI = constrainf(throttleI, -1.0f * GPS_RESCUE_MAX_THROTTLE_ITERM, 1.0f * GPS_RESCUE_MAX_THROTTLE_ITERM);
-    // up to 20% increase in throttle from I alone, need to check if this is needed, in practice.
-
-    // D component
-    const float throttleD = getAltitudeDerivative() * getAltitudePidCoeffs()->Kd * rescueState.intent.throttleDMultiplier;
-
-    // F component
-    // add a feedforward element that is proportional to the ascend or descend rate
-    const float throttleF = getAltitudePidCoeffs()->Kf * rescueState.intent.targetAltitudeStepCm * TASK_GPS_RESCUE_RATE_HZ;
-
-    const float hoverOffset = positionControlConfig()->hover_throttle - PWM_RANGE_MIN;
 
-    const float tiltMultiplier = 2.0f - fmaxf(getCosTiltAngle(), 0.5f); // same code as alt_hold
-    // 1 = flat, 1.24 at 40 degrees, max 1.5 around 60 degrees, the default limit of Angle Mode
-    // 2 - cos(x) is between 1/cos(x) and 1/sqrt(cos(x)) in this range
-
-    rescueThrottle = PWM_RANGE_MIN + (throttleP + throttleI - throttleD + throttleF + hoverOffset) * tiltMultiplier;
-
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 2, lrintf(rescueThrottle)); // normal range 1000-2000 but is before constraint
-
-    // constrain rescue throttle
-    rescueThrottle = constrainf(rescueThrottle, gpsRescueConfig()->throttleMin, gpsRescueConfig()->throttleMax);
-
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 3, lrintf(tiltMultiplier * 100));
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 4, lrintf(throttleP));
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 5, lrintf(throttleI));
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 6, lrintf(throttleD));
-    DEBUG_SET(DEBUG_GPS_RESCUE_THROTTLE_PID, 7, lrintf(throttleF));
-    DEBUG_SET(DEBUG_GPS_RESCUE_TRACKING, 6, lrintf(rescueThrottle));         // throttle value to use during a rescue
+    const float verticalVelocity = -getAltitudeDerivative() * rescueState.intent.throttleDMultiplier;
```

Comment:
Why is velocity negated here?

Reply:
Yes my mistake there, thank you.
I fixed it in AltHold while updating, but forgot to do so here.
Good catch, thank you.